src/Sarif.Viewer.VisualStudio/Fixes/FixSuggestedActionsSource.cs

```diff
@@ -97,7 +97,7 @@ public async Task<bool> HasSuggestedActionsAsync(ISuggestedActionCategorySet req
         {
             await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            return GetSuggestedActions(requestedActionCategories, range, cancellationToken).Any();
+            return GetSuggestedActions(requestedActionCategories, range, cancellationToken)?.Any() == true;
```

Comment:
> == true [](start = 98, length = 8)

is this direct comparison to true required to use the ? operator in the way that you have? if not, suggest deleting it.

Reply:
Yes, the data type of `?.Any()` is `Nullable<bool>`, which can't be directly compared to `bool`.

---
In reply to: [520858010](https://github.com/microsoft/sarif-visualstudio-extension/pull/265#discussion_r520858010) [](ancestors = 520858010)

Reply:
I would have used .Value to resolve this rather than depending on the implicit type conversion. In general, direct comparisons to boolean literals look odd (and are often frowned against in style guidelines).

---
In reply to: [520862773](https://github.com/microsoft/sarif-visualstudio-extension/pull/265#discussion_r520862773) [](ancestors = 520862773,520858010)

Reply:
If a `Nullable<bool>` is `null` you can't use `.Value` on it. Explicit Boolean comparisons on `Nullable<bool>` is really the only way to do it.

---
In reply to: [520864148](https://github.com/microsoft/sarif-visualstudio-extension/pull/265#discussion_r520864148) [](ancestors = 520864148,520862773,520858010)